artifactory/utils/storageinfo.go

```diff
@@ -71,3 +73,78 @@ func (sim *StorageInfoManager) GetRepoSummary(repoKey string) (*utils.Repository
 	}
 	return retVal, err
 }
+
+// GetReposTotalSize gets the total size of all passed repositories, in bytes.
+// This method must be called after CalculateStorageInfo.
+// The result of this function might not be accurate!
+func (sim *StorageInfoManager) GetReposTotalSize(repoKeys ...string) (int64, error) {
+	var totalSize int64
+	reposCounted := 0
+	pollingExecutor := &httputils.PollingExecutor{
+		Timeout:         getRepoSummaryPollingTimeout,
+		PollingInterval: getRepoSummaryPollingInterval,
+		MsgPrefix:       "Waiting for storage info calculation completion",
+		PollingAction: func() (shouldStop bool, responseBody []byte, err error) {
+			storageInfo, err := sim.GetStorageInfo()
+			if err != nil {
+				return true, nil, err
+			}
+			reposMap := make(map[string]bool)
+			for _, repoKey := range repoKeys {
+				reposMap[repoKey] = true
+			}
+			for _, repoSummary := range storageInfo.RepositoriesSummaryList {
+				if reposMap[repoSummary.RepoKey] {
+					reposCounted++
+					if repoSummary.UsedSpaceInBytes.String() != "" {
+						sizeToAdd, err := repoSummary.UsedSpaceInBytes.Int64()
+						if err != nil {
```

Comment:
Not sure I understand this line - 
If we received an error, we append the value and continue? What is the reason for this?

Reply:
My mistake. It should be `if err == nil...`